vms/evm/subscriber.go

```diff
@@ -161,58 +162,78 @@ func (s *subscriber) ProcessFromHeight(height *big.Int) error {
 		return err
 	}
 
-	// Only process logs if the provided height is not nil. Otherwise, simply update the database with
-	// the latest block height
-	if height != nil {
-		// Cap the number of blocks to process to MaxBlocksToProcess
-		toBlock := big.NewInt(0).SetUint64(latestBlock)
-		if height.Cmp(big.NewInt(0).Add(toBlock, big.NewInt(-MaxBlocksToProcess))) < 0 {
-			s.logger.Warn(
-				fmt.Sprintf("Requested to process too many blocks. Processing only the most recent %s blocks", MaxBlocksToProcess),
-				zap.String("requestedBlockHeight", height.String()),
-				zap.String("latestBlockHeight", toBlock.String()),
-			)
-			height = big.NewInt(0).Add(toBlock, big.NewInt(-MaxBlocksToProcess))
-		}
+	// Cap the number of blocks to process to MaxBlocksToProcess
+	toBlock := big.NewInt(0).SetUint64(latestBlock)
```

Comment:
nit: can we use `new(big.Int)` instead of `big.NewInt(0)`

Reply:
I'd prefer to use the package's constructor rather than a call to `new` directly.

Reply:
sgtm no strong pref here. Though is that the standard we should use throughout repos? I'm assuming you prefer the constructor incase it includes some other initialization that `new(big.Int)` wouldn't?

Reply:
Yes, in general we should initialize objects using constructors when available, as that leaves any initialization logic up to the type. If no constructor is available, and there are fields to initialize, we should prefer `&T{}` to get the pointer to an object. Only in cases in which we don't want to initialize _anything_ (such as an empty map) should we use `new` and `make`, as they only allocate and do not initialize. Exceptions to these rules are of course possible.